src/components/checkbox/demoSelectAll/script.js

```diff
@@ -0,0 +1,41 @@
+
+angular.module('checkboxDemo3', ['ngMaterial'])
+
+.controller('AppCtrl', function($scope) {
+  $scope.items = [1,2,3,4,5];
+  $scope.selected = [1];
+  $scope.toggle = function (item, list) {
+    var idx = list.indexOf(item);
+    if (idx > -1) list.splice(idx, 1);
+    else list.push(item);
+  };
+
+  $scope.exists = function (item, list) {
+    return list.indexOf(item) > -1;
+  };
+
+  $scope.isIndeterminate = function() {
+    if ($scope.selected.length !== 0 &&
+        $scope.selected.length !== $scope.items.length) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
This check can be replaced with `return $scope.selected.length !== 0 && $scope.selected.length !== $scope.items.length;`.